src/Data/ProgrammesDb/EntityRepository/CoreEntityRepository.php

```diff
@@ -196,6 +196,30 @@ public function findByPidFull(string $pid, string $entityType = 'CoreEntity'): ?
         return $this->resolveCategories($withHydratedParents)[0];
     }
 
+    /**
+     * @param string[] $pids
+     * @param string $entityType
+     * @return array
+     */
+    public function findByPids(array $pids, string $entityType = 'CoreEntity'): array
+    {
+        $this->assertEntityType($entityType, self::ALL_VALID_ENTITY_TYPES);
+        $qb = $this->getEntityManager()->createQueryBuilder()
+            ->select(['entity', 'image', 'masterBrand', 'network', 'mbImage', 'category', 'nwImage'])
+            ->from('ProgrammesPagesService:' . $entityType, 'entity') // For filtering on type
+            ->leftJoin('entity.image', 'image')
+            ->leftJoin('entity.masterBrand', 'masterBrand')
+            ->leftJoin('masterBrand.image', 'mbImage')
+            ->leftJoin('masterBrand.network', 'network')
+            ->leftJoin('network.image', 'nwImage')
+            ->leftJoin('entity.categories', 'category')
+            ->andWhere('entity.pid IN (:pids)')
```

Comment:
Do we want to limit the number of pids that can be passed in? I think the most you can have is 65535.

Reply:
I don't think it needs to be explicitly limited. If somebody's passing that many PIDs here, that's already going to make the page unusably slow, anything like that should be caught in code review.